### PR TITLE
[3.2] Add /v1/chain/get_unapplied_transactions to producer_api_plugin

### LINKED ISSUE #109: Allow inspection of various queues within nodeos
On a block producing (and other nodes), there are various transactions queued up for various purposes. Pending queue for example on A BP node.

Add producer APIs to allow the inspection of these kind of queues.

Show the transaction id, authorizing account, cpu usage and other interesting things.

This request is vague as I am not sure what information is easy to obtain.  The goal is to provide node operators details to manage situation where their nodes are under heavy CPU load and can easily find which account/contract is heavy user and possibly take actions to mitigate the impact of high levels of "abuse".

### PR DESCRIPTION
**Pull request by heifner**
Add `/v1/chain/get_unapplied_transactions` to `producer_api_plugin`

Interface:
```
   get_unapplied_transactions_params {
      string                lower_bound;  /// transaction id
      optional<uint32_t>    limit = 100;
      optional<uint32_t>    time_limit_ms; // defaults to 10ms
   };

   unapplied_trx {
      transaction_id_type       trx_id;
      time_point_sec            expiration;
      string                    trx_type; // eosio::chain::trx_enum_type values or "read_only"
      account_name              first_auth;
      account_name              first_receiver;
      action_name               first_action;
      uint16_t                  total_actions = 0;
      uint32_t                  billed_cpu_time_us = 0;
      size_t                    size = 0;
   };

   get_unapplied_transactions_result {
      size_t                     size = 0;
      size_t                     incoming_size = 0;
      vector<unapplied_trx>      trxs;
      string                     more; ///< fill lower_bound with trx id to fetch next set of transactions
   };
```

Resolves #109 

Example use:
```
$ curl -s http://127.0.0.1:8888/v1/producer/get_unapplied_transactions -d '{"limit":2}' | jq
{
  "size": 7495,
  "incoming_size": 0,
  "trxs": [
    {
      "trx_id": "fa9692a4bdd688bf889890d972e1d3a2ee00ad636bdd7ce56115ac962760c0d9",
      "expiration": "2022-09-17T16:31:10",
      "trx_type": "aborted",
      "first_auth": "fhjnm.c.wam",
      "first_receiver": "m.federation",
      "first_action": "mine",
      "total_actions": 1,
      "billed_cpu_time_us": 701,
      "size": 934
    },
    {
      "trx_id": "731fc1f6c1e7ed958090fad622cf4f11af97bab44320a42876c38703faadbb59",
      "expiration": "2022-09-17T16:30:16",
      "trx_type": "aborted",
      "first_auth": "a44cg.wam",
      "first_receiver": "m.federation",
      "first_action": "mine",
      "total_actions": 1,
      "billed_cpu_time_us": 746,
      "size": 934
    }
  ],
  "more": "dcd3464938ffc06f8aced1a1706ae0082fdaabc680f58854bfb8932ed822ce2c"
}
$ curl -s http://127.0.0.1:8888/v1/producer/get_unapplied_transactions -d '{"lower_bound":"dcd3464938ffc06f8aced1a1706ae0082fdaabc680f58854bfb8932ed822ce2c","limit":2}' | jq
{
  "size": 12428,
  "incoming_size": 4475,
  "trxs": [
    {
      "trx_id": "dcd3464938ffc06f8aced1a1706ae0082fdaabc680f58854bfb8932ed822ce2c",
      "expiration": "2022-09-17T16:30:16",
      "trx_type": "aborted",
      "first_auth": "jkbsg.wam",
      "first_receiver": "m.federation",
      "first_action": "mine",
      "total_actions": 1,
      "billed_cpu_time_us": 504,
      "size": 934
    },
    {
      "trx_id": "0b1e96b1eb56855d4eeee9f70ca9c623296212304851743c916b65e52ff83999",
      "expiration": "2022-09-17T16:30:16",
      "trx_type": "aborted",
      "first_auth": "kxiik.wam",
      "first_receiver": "m.federation",
      "first_action": "mine",
      "total_actions": 1,
      "billed_cpu_time_us": 545,
      "size": 934
    }
  ],
  "more": "00edb80be7eb903d74970d653cb63489c573d55d867078b794328e13c32e3227"
}
```

### CODE REVIEW

**Comment by ericpassmore** [comment]
Updated openapi updates producer.swagger.yaml
fixes #184

File: plugins/producer_plugin/producer_plugin.cpp [hunk 1]
```diff
@@ -1438,6 +1438,78 @@ producer_plugin::get_account_ram_corrections( const get_account_ram_corrections_
...
+            return "incoming";
+      }
+      return "unknown type";
+   };
+
+   get_unapplied_transactions_result result;
+   result.size = ua.size();
+   result.incoming_size = ua.incoming_size();
+
+   uint32_t remaining = p.limit ? *p.limit : std::numeric_limits<uint32_t>::max();
```
Comment (ericpassmore) [inline plugins/producer_plugin/producer_plugin.cpp:1483]: Defaulting to max<unit32> seems big. Not sure what is typical, I would have made the default a max of short int, uint16. Max should be bigger up to unit32.

Comment (heifner) [reply plugins/producer_plugin/producer_plugin.cpp:1483]: It is limited by the `time_limit_ms`, setting `limit` to max uint32 just means it is limited by time only.

**Review by ericpassmore — APPROVED** [review]

**Review by spoonincode — CHANGES_REQUESTED** [review]
ah one sec

**Review by ndcgundlach — APPROVED** [review]

**Pull request merged**